src/backend/cuda/scan.cpp

```diff
@@ -22,20 +22,10 @@ template<af_op_t op, typename Ti, typename To>
 Array<To> scan(const Array<Ti>& in, const int dim, bool inclusive_scan) {
     Array<To> out = createEmptyArray<To>(in.dims());
 
-    if (inclusive_scan) {
-        switch (dim) {
-            case 0: kernel::scan_first<Ti, To, op, true>(out, in); break;
-            case 1: kernel::scan_dim<Ti, To, op, 1, true>(out, in); break;
-            case 2: kernel::scan_dim<Ti, To, op, 2, true>(out, in); break;
-            case 3: kernel::scan_dim<Ti, To, op, 3, true>(out, in); break;
-        }
+    if (dim == 0) {
+        kernel::scan_first<Ti, To, op>(out, in, inclusive_scan);
```

Comment:
Op can also be a function argument can't it?

Reply:
It can, but the problem is larger than it seems to be. Lot of these template parameters,especially the ones in `src/backend/[cuda|opencl|cpu]/*.cpp` level, trickle down from `src/api/c` level. Even though we can now have `op` as normal argument to reduce template instantiations in CUDA backend alone, that will effect the function signatures at `src/backend/[cuda|opencl|cpu]/*.hpp` level. Unless we change all the headers in every other backend, it will become head ache.

Having said that, since it is already a template paramater when it reaches cuda backend implementation level, we might as well use it untill we change the backend API too reduce number of template parameters.

Reply:
Its not something you need to do in this PR. We can change that later.